cpp/src/arrow/tensor_test.cc

```diff
@@ -152,6 +243,22 @@ TEST(TestTensor, MakeFailureCases) {
   // negative items in shape
   ASSERT_RAISES(Invalid, Tensor::Make(float64(), data, {-3, 6}));
 
+  // overflow in stride computation
+  constexpr uint64_t total_length =
+      1 + static_cast<uint64_t>(std::numeric_limits<int64_t>::max());
+  EXPECT_RAISES_WITH_MESSAGE_THAT(
+      Invalid,
+      testing::HasSubstr(
+          "Row-major strides computed from shape would not fit in 64-bit integer"),
+      Tensor::Make(float64(), data, {2, 2, static_cast<int64_t>(total_length / 4)}));
+
+  // overflow by negative multiplication in strides validity check
+  EXPECT_RAISES_WITH_MESSAGE_THAT(
+      Invalid, testing::HasSubstr("would not fit in 64-bit integer"),
+      Tensor::Make(
+          float64(), data, {0, 4, 0},
+          {sizeof(double), -static_cast<int64_t>(total_length / 2), sizeof(double)}));
```

Comment:
What if, for example, `shape = {3, total_length / 3}` and `strides = {-total_length / 3, -1}`?

Reply:
I replaced this assertion with the new one to check that negative strides are prohibited.